packages/firestore/src/api/user_data_reader.ts

```diff
@@ -50,6 +50,29 @@ import { DocumentReference } from './database';
 
 const RESERVED_FIELD_REGEX = /^__.*__$/;
 
+/**
+ * An untyped Firestore Data Converter interface that is shared between the
+ * lite, full and legacy SDK.
+ */
+export interface UntypedFirestoreDataConverter<T> {
+  toFirestore(modelObject: T): firestore.DocumentData;
+  fromFirestore(snapshot: unknown, options?: unknown): T;
+}
+
+/**
+ * A reference to a document in a Firebase project.
+ *
+ * This class serves as a common base class for the public DocumentReferences
+ * exposed in the lite, full and legacy SDK.
+ */
+export class DocumentKeyReference<T> {
```

Comment:
This is the DocumentKeyReference type from the other PR, but it takes an `UntypedFirestoreDataConverter` so it can be shared. While that's a bit of a hack, it might be okay since it is purely a user type that we don't rely on internally.

Reply:
I merged in master.